components/dashboard/src/components/SelectIDEComponent.tsx

```diff
@@ -0,0 +1,145 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { IDEOption, IDEOptions } from "@gitpod/gitpod-protocol/lib/ide-protocol";
+import { useEffect, useMemo, useState } from "react";
+import { getGitpodService } from "../service/service";
+import { DropDown2, DropDown2Element } from "./DropDown2";
+import Editor from "../icons/Editor.svg";
+
+interface SelectIDEComponentProps {
+    selectedIdeOption?: string;
+    useLatest?: boolean;
+    onSelectionChange: (ide: string, latest: boolean) => void;
+}
+
+export default function SelectIDEComponent(props: SelectIDEComponentProps) {
+    const [ideOptions, setIdeOptions] = useState<IDEOptions>();
+    useEffect(() => {
+        getGitpodService().server.getIDEOptions().then(setIdeOptions);
+    }, []);
+    const getElements = useMemo(() => {
+        return (search: string) => {
+            if (!ideOptions) {
+                return [];
+            }
+            const options = IDEOptions.asArray(ideOptions);
+            const result: DropDown2Element[] = [];
+            for (const ide of options.filter(
+                (ide) =>
+                    `${ide.label}${ide.title}${ide.notes}${ide.id}`.toLowerCase().indexOf(search.toLowerCase()) !== -1,
+            )) {
+                result.push({
+                    id: ide.id,
+                    element: <IdeOptionElementInDropDown option={ide} useLatest={false} />,
+                    isSelectable: true,
+                });
+                if (ide.imageVersion !== ide.latestImageVersion) {
+                    result.push({
+                        id: ide.id + "-latest",
+                        element: <IdeOptionElementInDropDown option={ide} useLatest={true} />,
+                        isSelectable: true,
+                    });
+                }
+            }
+            return result;
+        };
+    }, [ideOptions]);
+    const internalOnSelectionChange = (id: string) => {
+        const { ide, useLatest } = parseId(id);
+        props.onSelectionChange(ide, useLatest);
+    };
+    const ide = props.selectedIdeOption || ideOptions?.defaultIde || "";
+    return (
+        <DropDown2
+            getElements={getElements}
+            onSelectionChange={internalOnSelectionChange}
+            searchPlaceholder={"Select Editor"}
+        >
+            <IdeOptionElementSelected option={ideOptions?.options[ide]} useLatest={!!props.useLatest} />
+        </DropDown2>
+    );
+}
+
+function parseId(id: string): { ide: string; useLatest: boolean } {
+    const useLatest = id.endsWith("-latest");
+    const ide = useLatest ? id.slice(0, -7) : id;
+    return { ide, useLatest };
+}
+
+interface IdeOptionElementProps {
+    option: IDEOption | undefined;
+    useLatest: boolean;
+}
+
+function IdeOptionElementSelected(p: IdeOptionElementProps): JSX.Element {
```

Comment:
**Nit/Opinion:** A couple alternatives to typing a component like this that can save the destructuring of props at the beginning

```ts
function IdeOptionElementSelected({ option, useLatest }: IdeOptionElementProps): JSX.Element {
```

I personally tend to prefer the following style, mostly because it abstracts the return type a bit for all of the values React supports for returning (i.e. allows a `return null`)

```ts
const IdeOptionElementSelected: FunctionComponent<IdeOptionElementProps> = ({ option, useLatest }) => {
```